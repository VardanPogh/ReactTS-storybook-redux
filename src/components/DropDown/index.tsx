import React from 'react'
import { createStyles, makeStyles, Theme } from '@material-ui/core/styles'
import InputLabel from '@material-ui/core/InputLabel'
import MenuItem from '@material-ui/core/MenuItem'
import FormHelperText from '@material-ui/core/FormHelperText'
import FormControl from '@material-ui/core/FormControl'
import Select from '@material-ui/core/Select'

const useStyles = makeStyles((theme: Theme) =>
  createStyles({
    formControl: {
      margin: theme.spacing(1),
      minWidth: 120,
    },
    helperText: {
      marginLeft: 1,
    },
  })
)

interface Items {
  value: string
  label: string
}

interface Props {
  id: string
  label?: string
  value: string
  onChange(event: React.ChangeEvent<{ value: unknown }>): void
  hasHelper?: boolean
  helperText?: string
  items: Array<Items>
}

const DropDown: React.FC<Props> = ({
  id,
  label,
  value,
  onChange,
  hasHelper,
  helperText,
  items,
}) => {
  const classes = useStyles()

  return (
    <FormControl variant="outlined" className={classes.formControl}>
      <InputLabel id={id}>{label}</InputLabel>
      <Select
        labelId={id}
        id={id}
        value={value}
        onChange={onChange}
        label={label}
      >
        {items.map((item, index) => (
          <MenuItem key={index} value={item.value}>
            {item.label}
          </MenuItem>
        ))}
      </Select>
      {hasHelper && (
        <FormHelperText className={classes.helperText}>
          {helperText}
        </FormHelperText>
      )}
    </FormControl>
  )
}

export default DropDown
