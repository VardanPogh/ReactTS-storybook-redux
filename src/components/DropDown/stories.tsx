import React, { useState } from 'react'
import { withKnobs, text, object, boolean } from '@storybook/addon-knobs'

import DropDown from '.'

export default {
  title: 'DropDown',
  component: DropDown,
  decorators: [withKnobs],
}

const items = [
  { value: '', label: 'None' },
  { value: '10', label: 'Ten' },
  { value: '20', label: 'Twenty' },
  { value: '30', label: 'Thirty' },
]

export const Default: React.FC = () => {
  const [age, setAge] = useState('')

  const handleChange = (event: React.ChangeEvent<{ value: unknown }>): void => {
    setAge(event.target.value as string)
  }
  return (
    <DropDown
      id={text('Id', 'test')}
      label={text('Label', 'test')}
      value={age}
      onChange={handleChange}
      hasHelper={boolean('Has helper', true)}
      helperText={text('Helper Text', 'helper text')}
      items={object('Items', items)}
    />
  )
}
