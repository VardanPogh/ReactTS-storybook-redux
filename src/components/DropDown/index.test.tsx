import React from 'react'
import { render } from '@testing-library/react'

import DropDown from '.'

describe('DropDown Test', () => {
  test('Load Button', () => {
    const myMock = jest.fn()
    const items = [
      { value: '', label: 'None' },
      { value: '10', label: 'Ten' },
      { value: '20', label: 'Twenty' },
      { value: '30', label: 'Thirty' },
    ]

    const { container } = render(
      <DropDown
        id="test"
        label="test"
        value="10"
        onChange={myMock}
        hasHelper
        helperText="Helper Text"
        items={items}
      />
    )
    expect(container).toMatchSnapshot()
  })
})
