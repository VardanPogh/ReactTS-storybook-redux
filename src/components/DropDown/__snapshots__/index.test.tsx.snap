// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DropDown Test Load Button 1`] = `
<div>
  <div
    class="MuiFormControl-root makeStyles-formControl-1"
  >
    <label
      class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-filled"
      data-shrink="true"
      id="test"
    >
      test
    </label>
    <div
      class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
    >
      <div
        aria-haspopup="listbox"
        aria-labelledby="test test"
        class="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input"
        id="test"
        role="button"
        tabindex="0"
      >
        Ten
      </div>
      <input
        aria-hidden="true"
        class="MuiSelect-nativeInput"
        tabindex="-1"
        value="10"
      />
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSelect-icon MuiSelect-iconOutlined"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M7 10l5 5 5-5z"
        />
      </svg>
      <fieldset
        aria-hidden="true"
        class="PrivateNotchedOutline-root-3 MuiOutlinedInput-notchedOutline"
      >
        <legend
          class="PrivateNotchedOutline-legendLabelled-5 PrivateNotchedOutline-legendNotched-6"
        >
          <span>
            test
          </span>
        </legend>
      </fieldset>
    </div>
    <p
      class="MuiFormHelperText-root MuiFormHelperText-contained makeStyles-helperText-2 MuiFormHelperText-filled"
    >
      Helper Text
    </p>
  </div>
</div>
`;
